src/gilded_rose.js

```diff
@@ -13,50 +13,94 @@ items.push(new Item('Sulfuras, Hand of Ragnaros', 0, 80));
-          items[i].quality = items[i].quality - 1
-        }
-      }
-    } else {
-      if (items[i].quality < 50) {
-        items[i].quality = items[i].quality + 1
-        if (items[i].name == 'Backstage passes to a TAFKAL80ETC concert') {
-          if (items[i].sell_in < 11) {
-            if (items[i].quality < 50) {
-              items[i].quality = items[i].quality + 1
-            }
-          }
-          if (items[i].sell_in < 6) {
-            if (items[i].quality < 50) {
-              items[i].quality = items[i].quality + 1
-            }
-          }
-        }
-      }
-    }
-    if (items[i].name != 'Sulfuras, Hand of Ragnaros') {
-      items[i].sell_in = items[i].sell_in - 1;
-    }
-    if (items[i].sell_in < 0) {
-      if (items[i].name != 'Aged Brie') {
-        if (items[i].name != 'Backstage passes to a TAFKAL80ETC concert') {
-          if (items[i].quality > 0) {
-            if (items[i].name != 'Sulfuras, Hand of Ragnaros') {
-              items[i].quality = items[i].quality - 1
-            }
-          }
-        } else {
-          items[i].quality = items[i].quality - items[i].quality
+  // for (var i = 0; i < items.length; i++) {
+  //   if (items[i].name != 'Aged Brie' && items[i].name != 'Backstage passes to a TAFKAL80ETC concert') {
+  //     if (items[i].quality > 0) {
+  //       if (items[i].name != 'Sulfuras, Hand of Ragnaros') {
+  //         items[i].quality = items[i].quality - 1
+  //       }
+  //     }
+  //   } else {
+  //     if (items[i].quality < 50) {
+  //       items[i].quality = items[i].quality + 1
+  //       if (items[i].name == 'Backstage passes to a TAFKAL80ETC concert') {
+  //         if (items[i].sell_in <= 11) { // Made change from <  to <=
+  //           if (items[i].quality < 50) {
+  //             items[i].quality = items[i].quality + 1
+  //           }
+  //         }
+  //         if (items[i].sell_in <= 6) { // Made change from <  to <=
+  //           if (items[i].quality < 50) {
+  //             items[i].quality = items[i].quality + 1
+  //           }
+  //         }
+  //       }
+  //     }
+  //   }
+  //   if (items[i].name != 'Sulfuras, Hand of Ragnaros') {
+  //     items[i].sell_in = items[i].sell_in - 1;
+  //   }
+  //   if (items[i].sell_in < 0) {
+  //     if (items[i].name != 'Aged Brie') {
+  //       if (items[i].name != 'Backstage passes to a TAFKAL80ETC concert') {
+  //         if (items[i].quality > 0) {
+  //           if (items[i].name != 'Sulfuras, Hand of Ragnaros') {
+  //             items[i].quality = items[i].quality - 1
+  //           }
+  //         }
+  //       } else {
+  //         items[i].quality = items[i].quality - items[i].quality
+  //       }
+  //     } else {
+  //       if (items[i].quality < 50) {
+  //         items[i].quality = items[i].quality + 1
+  //       }
+  //     }
+  //   }
+  // }
+var numUpdateby = 1
+  items.forEach((index) => {
+    index.sell_in -=1
+    switch(index.name)
```

Comment:
At the time you realize you have too many `if`'s in your code and you need another logic structure, changing to `switch` is an easy step but `switch` is just as bad as multiple `if`'s.

Explore other ways to refactor your code to avoid `if`'s and `siwtch` statements. `map`'s and design patterns are better alternatives because they are easier to extend and reason about.